snakemake/shell.py

```diff
@@ -57,6 +66,12 @@ def executable(cls, cmd):
                 )
         if os.path.split(cmd)[-1] == "bash":
             cls._process_prefix = "set -euo pipefail; "
+        if ON_WINDOWS and cmd.lower().endswith("bash.exe"):
+            # If bash is selected as the executable on Windows
+            # configure _preocess_prefix and win_quote_cmd for
+            # this to work correctly
+            cls._process_prefix = '"{}" -c'.format(cmd)
```

Comment:
The `cls._process_args["executable"]` is passed to Popen as the `executable` parameter. That doesn't do anything on Windows. So if we need to use bash instead of cmd.exe we need to pass use `bash.exe -c ` as part of the command. The `cls._process_prefix` is perfect for that. I still let code set bash.exe into the executable argument so it doesn't differ for from posix.